development/gcbuild/main.go

```diff
@@ -0,0 +1,340 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/kyma-project/test-infra/development/gcbuild/tags"
+	"gopkg.in/yaml.v2"
+	"io"
+	"io/ioutil"
+	errutil "k8s.io/apimachinery/pkg/util/errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+type Config struct {
+	Steps         []Step            `yaml:"steps"`
+	Substitutions map[string]string `yaml:"substitutions"`
+	Images        []string          `yaml:"images"`
+}
+
+func getConfig(f string) (*Config, error) {
+	b, err := ioutil.ReadFile(f)
+	if err != nil {
+		return nil, err
+	}
+	var cb Config
+	if err := yaml.Unmarshal(b, &cb); err != nil {
+		return nil, fmt.Errorf("cloudbuild.yaml parse error: %w", err)
+	}
+	return &cb, nil
+}
+
+type Step struct {
+	Name string   `yaml:"name"`
+	Args []string `yaml:"args"`
+}
+
+type options struct {
+	buildDir      string
+	configFile    string
+	variantsFile  string
+	variant       string
+	logDir        string
+	devRegistry   string
+	project       string
+	stagingBucket string
+	logsBucket    string
+	silent        bool
+	isCI          bool
+}
+
+// parseVariable returns a gcloud compatible substitution option.
+// Keys are set to upper-case and prefix "_" is set if not present.
+func parseVariable(key, value string) string {
+	k := strings.TrimSpace(strings.ToUpper(key))
+	if !strings.HasPrefix(k, "_") {
+		k = "_" + k
+	}
+	return k + "=" + strings.TrimSpace(value)
+}
+
+func getSHAFromGit() (string, error) {
+	cmd := exec.Command("git", "rev-parse", "head")
+	out, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	validTagRegexp, err := regexp.Compile("[^-_.a-zA-Z0-9]+")
+	if err != nil {
+		return "", err
+	}
+	sanitizedOutput := validTagRegexp.ReplaceAllString(string(out), "")
```

Comment:
What this step is doing? What you are sanitizing here?

Reply:
I yanked it from k8s tool. This ensures that she includes only allowed characters defined by regex. This costs nothing, and cleans the output of the `git` command if something unusual pops up.